src/Point/PointToInterlockingConnectionB4R1Impl.cs

```diff
@@ -0,0 +1,153 @@
+        await SendMessage(startInitialization);
+
+        var pointState = new B4R1PointStateImpl(state);
+        var initialPosition = new PointPointPositionMessage(_localId, _remoteId, pointState.PointPosition, pointState.DegradedPointPosition);
+        await SendMessage(initialPosition);
+
+        var completeInitialization = new PointInitialisationCompletedMessage(_localId, _remoteId);
+        await SendMessage(completeInitialization);
+        return true;
+    }
+
+    public async Task SendPointPosition(PointState state)
+    {
+        var pointState = new B4R1PointStateImpl(state);
+        var response = new PointPointPositionMessage(_localId, _remoteId, pointState.PointPosition, pointState.DegradedPointPosition);
+        await SendMessage(response);
+    }
+
+    async public Task SendTimeoutMessage()
+    {
+        var response = new PointTimeoutMessage(_localId, _remoteId);
+        await SendMessage(response);
+    }
+
+    public async Task<PointPosition?> ReceivePointPosition()
+    {
+        var message = await ReceiveMessage<PointMovePointCommand>();
+
+        return (message != null)? message.CommandedPointPosition switch
+        {
+            PointMovePointCommandCommandedPointPosition.SubsystemElectronicInterlockingRequestsARightHandPointMoving => PointPosition.RIGHT,
+            PointMovePointCommandCommandedPointPosition.SubsystemElectronicInterlockingRequestsALeftHandPointMoving => PointPosition.LEFT,
+        } : null;
+    }
+
+    public void Dispose()
+    {
+        _currentConnection?.Dispose();
+    }
+
+    private async Task SendMessage(Message message)
+    {
+        if (_currentConnection == null) throw new NullReferenceException("Connection is null. Did you call Connect()?");
+        await _currentConnection.RequestStream.WriteAsync(new SciPacket() { Message = ByteString.CopyFrom(message.ToByteArray()) });
+    }
+
+    private async Task<T?> ReceiveMessage<T>() where T : Message
+    {
+        if (_currentConnection == null) throw new NullReferenceException("Connection is null. Did you call Connect()?");
+        if (!await _currentConnection.ResponseStream.MoveNext(_timeout.Token)) return null;
+        
+        var message = Message.FromBytes(_currentConnection.ResponseStream.Current.Message.ToByteArray());
+        if (message is not T)
+        {
+            _logger.LogError("Unexpected message: {}", message);
+            return null;
+        }
+        return message as T;
+    }
+}
+
+public class B4R1PointStateImpl
+{
+    private PointState _state;
+    public PointPointPositionMessageReportedPointPosition PointPosition { get => map(_state.PointPosition); }
+    public PointPointPositionMessageReportedDegradedPointPosition DegradedPointPosition { get => map(_state.DegradedPointPosition); }
+
+    public B4R1PointStateImpl(PointState state)
+    {
+        _state = state;
+    }
+
+    private PointPointPositionMessageReportedPointPosition map(PointPosition value) => value switch
```

Comment:
Please review the naming conventions https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/capitalization-conventions

I'd prefer if we don't perform method overloading by parameter type here, instead use different method names for mapping point positions and degraded point positions.